File: android/src/main/kotlin/dev/note11/flutter_naver_map/flutter_naver_map/model/exception/NFlutterException.kt
package dev.note11.flutter_naver_map.flutter_naver_map.model.exception

data class NFlutterException(
    val code: String,
    val message: String?,
) {
    fun toMessageable(): Map<String, Any?> {
        return mapOf(
            "code" to code,
            "message" to message,
        )
    }
}

File: ios/flutter_naver_map/Sources/flutter_naver_map/model/exception/NFlutterException.swift
struct NFlutterException {
    let code: String
    let message: String?
    
    init(code: String, message: String?) {
        self.code = code
        self.message = message
    }
    
    init(error: Error) {
        self.code = error._code.description
        self.message = error.localizedDescription
    }
    
    func toMessageable() -> [String: Any?] {
        return [
            "code": code,
            "message": message
        ]
    }
}
